window.c

```diff
@@ -78,6 +78,16 @@ static void xft_alloc_color(const win_env_t *e, const char *name, XftColor *col)
 	if (!XftColorAllocName(e->dpy, e->vis, e->cmap, name, col))
 		error(EXIT_FAILURE, 0, "Error allocating color '%s'", name);
 }
+
+static void win_init_bar_buf(win_t *win)
+{
+	enum { BAR_L_LEN = 512, BAR_R_LEN = 64 };
```

Comment:
Why an enum instead of a normal const variable/macro?

Also what's the benefit of this vs just making b.l.buf an array of size BAR_L_LEN + 3?

Reply:
> Why an enum instead of a normal const variable/macro?

Using a variable would turn the array into a VLA. And a macro has file scope while the enum is function scope.

> Also what's the benefit of this vs just making b.l.buf an array of size BAR_L_LEN + 3?

Either this or we hardcode the size into the array and then do `- 3` when assigning to the `.size` since the utf8 decoder needs 3 bytes of padding.

Reply:
>Using a variable would turn the array into a VLA.

Just assumed the compliler would be smart if we marked it consts... but probably shouldn't assume that. 

I'd say just hardcode it into the array. We could even do
```
struct bar:
{
   ...
   char lbuf[BAR_L_LEN];
   char utf8_padding_lbuf[3];
   char rbuf[BAR_R_LEN];
   char utf8_padding_rbuf[3];
}
```
Could event just inline BAR_L_LEN and use sizeof to get the size whenever we care. (And if we don't want the user to be able to change BAR_L_LEN/BAR_R_LEN in their config, why not just inline it?)

I don't hate this patch. I just would think it'd be better to not have an init function if we aren't really doing anything. I'd go so far to say that the above is clearer and more self-documenting than hiding these details in one of the many init functions. Here you can clearly see the constant size of the struct and the fact that we are accounting for utf8 padding

Reply:
Yeah, hardcoding the array size into the win struct is an option. But that'd be a more intrusive change and would require changing all instance of `l.buf` => `.lbuf` and `.size` => `sizeof/ARRLEN(...)`.

The current way seemed like the less intrusive way of achieving the goal. And for what it's worth, it'd also be easier to switch back to dynamic allocation later on if we want to.

So if the enum is the main point of objection, I'm okay removing it and hardcoding the size. And if the main problem is the init function, then I can remove the new function and inline what it's currently doing directly at `win_init()`.

Does that sound good?

Reply:
>.size => sizeof/ARRLEN(...).

Well if we assume sizof(char) == 1, then it'd just be .size -> sizeof

>The current way seemed like the less intrusive way of achieving the goal. And for what it's worth, it'd also be easier to switch back to dynamic allocation later on if we want to.

I'd feel like we'd never need dynamic allocation. We could make the bars large enough to fill the screen with text and not see a noticeable increase in memory used. I do agree though that my proposal has a larger diff and there isn't much of a practical difference either way.

>Does that sound good?

Yes to both.

Reply:
> Well if we assume sizof(char) == 1, then it'd just be .size -> sizeof

sizeof(char) == 1 is guaranteed, so using sizeof vs ARRLEN on a char array is mostly a matter of style.

> Yes to both.

Done at 1f1161f